src/Component.lua

```diff
@@ -341,7 +341,7 @@ function Component:__mount(reconciler, virtualNode)
 		instance:didMount()
 	end
 
-	if internalData.pendingState ~= nil then
+	if (internalData :: any).pendingState ~= nil then
```

Comment:
I'd suggest adding `pendingState = nil` to the `internalData` table declaration instead of this hard cast operator.